### ISSUE TITLE
sqlmap should not make headers uppercase when reading from file

### ISSUE THREAD

**Issue by lorenzog** [sqlmapproject/sqlmap#2379]
## What's the problem (or question)?

When using the `-r` option to read from a local file, sqlmap rewrites the headers and changes their case.

## Do you have an idea for a solution?

When parsing the headers from a file, make sure the 'key' part is not changed

## How can we reproduce the issue?

1. Create a file e.g.

```
POST /foo HTTP/1.1
Host: example.com
Accept: */*
x-my-header: 12345

some data goes here

```

2. Capture the traffic via tcpdump or by proxying it through burp
3. Observe the result:

```
POST /foo HTTP/1.1
Content-Length: 19
X-My-Header: 12345
Connection: close
Accept: */*
User-Agent: sqlmap/1.1#stable (http://sqlmap.org)
Host: example.com
Content-Type: application/x-www-form-urlencoded

some data goes here
```

1. Notice how `x-my-header` has been turned into `X-My-Header` (uppercase)

Note: the [spec](http://stackoverflow.com/a/5259004/204634) says that headers should not be case sensitive, but this does not mean every server out there adheres to it. 

Note: headers also are re-ordered, which might affect how sqlmap interacts with other servers.

## What are the running context details?

* Installation method (e.g. `pip`, `apt-get`, `git clone` or `zip`/`tar.gz`): apt-get
* Client OS (e.g. `Microsoft Windows 10`) Kali linux
* Program version (`python sqlmap.py --version` or `sqlmap --version` depending on installation): 1.1#stable
* Target DBMS (e.g. `Microsoft SQL Server`): n/a
* Detected WAF/IDS/IPS protection (e.g. `ModSecurity` or `unknown`): n/a
* SQLi techniques found by sqlmap (e.g. `error-based` and `boolean-based blind`):
* Results of manual target assessment (e.g. found that the payload `query=test' AND 4113 IN ((SELECT 'foobar'))-- qKLV` works): n/a
* Relevant console output (if any): n/a
* Exception traceback (if any): n/a

**Comment by stamparm** [comment]
1) `Note: headers also are re-ordered, which might affect how sqlmap interacts with other servers.` -> python does it because headers are stored internally inside the unordered dictionary
2) `Note: the spec says that headers should not be case sensitive, but this does not mean every server out there adheres to it.` -> you've explained the thing. Spec says one thing and you require something outer of specs without solid case. Can you please state one web platform that has issues with header name case sensitivity?

**Comment by lorenzog** [comment]
@stamparm regarding item 2. I've already encountered two web platforms that discriminate on header capitalization. One of them is Microsoft BizTalk; however I suspect that in both cases it might be due to some form of customisation. Nevertheless having sqlmap rewrite the headers is unnecessary and puts it "in the way". I had to tunnel its requests through a web proxy and use a macro to lowercase the header.

All I'm saying is that while the spec does not mandate it, given the nature of sqlmap and the fact that it's used as a testing tool it should not unnecessarily enforce behaviours such as this one.
